src/transit_data/data_init.rs

```diff
@@ -208,22 +359,84 @@ impl TransitData {
         let board_times = stop_times.clone().map(|s| s.board_time);
         let debark_times = stop_times.clone().map(|s| s.debark_time);
 
+        /*
+         * Flow correction with stay-in
+         *
+         * The stay-in section is allowed with 2 configurations:
+         *  - when two VJ share the same stop point with similar stop times (example 1)
+         *  - when two VJ are joined on 2 different stop points with consecutive stop times (example 2)
+         *
+         *   Example 1:
+         *   ----------
+         *         out          in   out         in
+         *          X    SP1    |    ▲    SP2    X
+         *          X           ▼    |           X
+         *    VJ:1   08:00-09:00      10:00-11:00
+         *    VJ:2                    10:00-11:00      14:00-15:00
+         *                           X           ▲    |           X
+         *                           X           |    ▼   SP3     X
+         *                           out         in   out         in
+         *                           |- Stay-In -|
+         *
+         *   Example 2:
+         *   ----------
+         *                                       (1)  (2)
+         *         out          in   out         in   out         in   out         in
+         *          X    SP1    |    ▲    SP2    |    ▲    SP3    |    ▲   SP4     X
+         *          X           ▼    |           ▼    |           |    |           X
+         *    VJ:1   08:00-09:00      10:00-11:00     |           ▼    |           X
+         *    VJ:2                                     12:00-13:00      14:00-15:00
+         *                           |---------- Stay In ---------|
+         *
+         *  Example 2 is the only case were we allow specific pick-up and drop-off
+         */
+        let has_prev_stay_in_on_same_stop =
+            match vehicle_journey_to_prev_stay_in.get(&vehicle_journey_idx) {
+                Some(StayInType::SameStopPoint(_)) => true,
+                Some(StayInType::DifferentStopPoint(_)) | None => false,
+            };
+        let has_next_stay_in_on_same_stop =
+            match vehicle_journey_to_next_stay_in.get(&vehicle_journey_idx) {
+                Some(StayInType::SameStopPoint(_)) => true,
+                Some(StayInType::DifferentStopPoint(_)) | None => false,
+            };
+
+        let nb_of_positions = flows.len();
+        let corrected_flows = flows.enumerate().map(|(position_idx, flow)| {
+            if position_idx == 0 && has_prev_stay_in_on_same_stop {
+                match flow {
+                    BoardAndDebark => BoardOnly,
+                    DebarkOnly => NoBoardDebark,
+                    _ => flow,
+                }
+            } else if position_idx == nb_of_positions - 1 && has_next_stay_in_on_same_stop {
+                match flow {
+                    BoardAndDebark => DebarkOnly,
+                    BoardOnly => NoBoardDebark,
+                    _ => flow,
```

Comment:
```suggestion
                    BoardAndDebark |  BoardOnly => BoardOnly,
                    DebarkOnly | NoBoardDebark => NoBoardDebark,
```
so we don't have suprises if we (ever) add another variant to the enum